spec/rubocop/cop/rspec/prefer_before_over_setup_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::PreferBeforeOverSetup do
+  context 'when using `do ... end` block syntax' do
+    it 'accepts before call' do
+      expect_no_offenses(<<~RUBY)
+        before do
+          allow(foo).to receive(:bar)
+        end
+      RUBY
+    end
+
+    it 'registers an offense on setup call' do
+      expect_offense(<<~RUBY)
+        setup do
+        ^^^^^^^^ Use `before` instead of `setup`.
+          allow(foo).to receive(:bar)
+        end
+      RUBY
+    end
+  end
+
+  context 'when using curly braces block syntax' do
+    it 'accepts before call' do
+      expect_no_offenses(<<~RUBY)
+        before {
+          allow(foo).to receive(:bar)
+        }
+      RUBY
+    end
+
+    it 'registers an offense on setup call' do
+      expect_offense(<<~RUBY)
+        setup {
+        ^^^^^^^ Use `before` instead of `setup`.
+          allow(foo).to receive(:bar)
+        }
+      RUBY
+    end
+  end
+
+  context 'when autocorrecting offenses' do
+    it 'replaces `setup` with `before`' do
+      expect(
+        autocorrect_source('setup do foo = 1 end')
```

Comment:
You may use `expect_correction(<<-RUBY)` that accepts a "good" source as its argument and uses the "bad" from the previously called `expect_offense`.